hypothesis-python/tests/cover/test_provisional_strategies.py

```diff
@@ -63,3 +68,18 @@ def test_valid_domains_arguments(max_length, max_element_length):
 @pytest.mark.parametrize("strategy", [domains(), urls()])
 def test_find_any_non_empty(strategy):
     find_any(strategy, lambda s: len(s) > 0)
+
+
+@given(_url_fragments_strategy)
+# There's a lambda in the implementation that only gets run if we generate at
+# least one percent-escape sequence, so we derandomize to ensure that coverage
+# isn't flaky.
+@settings(derandomize=True)
```

Comment:
I realized that my original test didn't actually reliably hit the lambda, so I'm setting `derandomize=True` to give us some consistency.

Reply:
I don't think it's reasonably possible to do better than this without changing the underlying implementation. The lambda only gets run if one of `st.characters` or `st.booleans` gives a favourable outcome.

(Extracting the inner strategy and adding a test gave us a much lower probability of flakiness, but didn't eliminate it entirely.)

Reply:
Now that I think about it, I could just get deterministic coverage by turning the lambda into a function, and testing that.

But having already written it this way, testing the lambda itself seems pretty artificial, so I don't mind leaving it as-is.